Sources/SwiftDocC/Semantics/Symbol/Symbol.swift

```diff
@@ -323,12 +332,75 @@ public final class Symbol: Semantic, Abstracted, Redirected, AutomaticTaskGroups
         self.redirectsVariants = redirectsVariants
         self.originVariants = originVariants
         self.automaticTaskGroupsVariants = automaticTaskGroupsVariants
-        self.extendedModule = extendedModule
     }
     
     public override func accept<V: SemanticVisitor>(_ visitor: inout V) -> V.Result {
         return visitor.visitSymbol(self)
     }
+
+    /**
+     Append a new generic constraint for the given extended module
+
+     - Parameters:
+        - extendedModule: Which module this symbol extends.
+        - typeKind: The ``SymbolGraph/Symbol/KindIdentifier`` of the symbol this extension extends.
+        - constraint: The new generic constraints to add
+        - trait: Which generic constraints variant to append to
+     - Returns: True if the constraint was appended to an existing or new swift extension variant. False if there was an existing swift extension variant for a different module.
+     */
+
+    public func addConstraint(
+        extendedModule: String,
+        typeKind: SymbolGraph.Symbol.KindIdentifier? = nil,
+        constraint newConstraint: SymbolGraph.Symbol.Swift.GenericConstraint,
+        trait: DocumentationDataVariantsTrait = DocumentationDataVariantsTrait.swift
```

Comment:
It's good that you are supporting multiple traits here but I'm wondering if we really need to. 

Considering that the added constraint is Swift specific, I think maybe we could remove the `trait` argument and rename the method to indicate that this is only applicable to Swift. That would allow for the implementation to ignore the other traits which realistically should never have this mixin. 

Maybe something like this?

```swift
    /// Append a new generic constraint for the given extended module
    /// - Parameters:
    ///    - extendedModule: The name of the extended module.
    ///    - extendedSymbolKind: The kind of the extended symbol.
    ///    - constraint: The new generic constraints to add.
    public func addSwiftExtensionConstraint(
        extendedModule: String,
        extendedSymbolKind: SymbolGraph.Symbol.KindIdentifier? = nil,
        constraint newConstraint: SymbolGraph.Symbol.Swift.GenericConstraint
    ) {
```